ovos_core/intent_services/commonqa_service.py

```diff
@@ -98,141 +126,158 @@ def match(self, utterances, lang, message):
+                LOG.debug(f"Waiting {time_to_wait}s before checking extensions")
+                query.responses_gathered.wait(time_to_wait)
             # not waiting for any more skills
-            if not self.query_extensions[search_phrase]:
-                self._query_timeout(message.reply('question:query.timeout',
-                                                  message.data))
-        else:
-            LOG.warning(f'{skill_id} Answered too slowly, will be ignored.')
-
-    def _query_timeout(self, message):
-        if not self.searching.is_set():
-            LOG.warning("got a common query response outside search window")
-            return  # not searching, ignore timeout event
-        self.searching.clear()
-
-        # Prevent any late-comers from retriggering this query handler
-        with self.lock:
-            LOG.info('Timeout occurred check responses')
-            search_phrase = message.data.get('phrase', "")
-            if search_phrase in self.query_extensions:
-                self.query_extensions[search_phrase] = []
-            self.enclosure.mouth_reset()
-
-            # Look at any replies that arrived before the timeout
-            # Find response(s) with the highest confidence
-            best = None
-            ties = []
-            if search_phrase in self.query_replies:
-                for handler in self.query_replies[search_phrase]:
-                    if not best or handler['conf'] > best['conf']:
-                        best = handler
-                        ties = []
-                    elif handler['conf'] == best['conf']:
-                        ties.append(handler)
-
-            if best:
-                if ties:
-                    # TODO: Ask user to pick between ties or do it automagically
-                    pass
-
-                # invoke best match
-                LOG.info('Handling with: ' + str(best['skill_id']))
-                if not message.data.get("handles_speech", False):
-                    self.speak(best['answer'])
-                cb = best.get('callback_data') or {}
-                self.bus.emit(message.forward('question:action',
-                                              data={'skill_id': best['skill_id'],
-                                                    'phrase': search_phrase,
-                                                    'callback_data': cb}))
-                self.answered = True
-            else:
-                self.answered = False
-            self.waiting = False
-            if search_phrase in self.query_replies:
-                del self.query_replies[search_phrase]
-            if search_phrase in self.query_extensions:
-                del self.query_extensions[search_phrase]
+            if not query.extensions:
+                LOG.debug(f"No more skills to wait for ({query.session_id})")
+                query.responses_gathered.set()
 
-    def speak(self, utterance, message=None):
-        """Speak a sentence.
```

Comment:
explanation for this removal
- skills loaded by core have ovos-workshop 0.0.15 minimum version
- since [V0.0.12a16](https://github.com/OpenVoiceOS/OVOS-workshop/tree/V0.0.12a16) (2023-04-22) skills are able to be run standalone, standalone skills have workshop >=0.0.12a16
- since the [V0.0.12a14](https://github.com/OpenVoiceOS/OVOS-workshop/tree/V0.0.12a14) (2023-04-22) common query base class does the speak

this means its impossible to load a skill in current core that doesnt emit the speak, migration is complete